azure-pipelines.yml

```diff
@@ -150,6 +153,9 @@ stages:
         - template: ".azure/test-linux.yml"
           parameters:
             pythonVersion: ${{ parameters.minimumPythonVersion }}
+            # Failures due to missing optionals should be very rare, so we
+            # optimise for faster turnaround in testing optional features.
+            installOptionals: true
```

Comment:
> Failures due to missing optionals should be very rare

What does that mean in a CI perspective? Why would missing optionals happen?

Reply:
"missing optionals" meant "the optional is not installed", which will be quite a common situation for users.  In the test suite, I'm assuming that it's going to be more common for a PR to cause a test case to fail due to a logic bug than it is to fail because there was a missing `@skipUnless` decorator on the test - _most_ PRs don't introduce new tests that need a skip decorator - so I put the "install all optionals == run all tests" in the weed-out stage, and left the "check that the test suite still works in a minimal install" version to the second stage.  The aim is that stage 1 should catch as many failures as possible using as few VMs / little walltime as possible to reduce our CI resource usage.

Reply:
Here's another possible wording for the comment:

> There are two failure cases with optionals: 1) in the optionals run, the tests that need optionals fail due to bugs. 2) in the no-optionals run, Terra fails because it is improperly expecting those optionals to be installed. The first case is more likely to result in issues, so we run it first in this initial stage of CI to get results faster.

Reply:
I expanded the comment in 4c0a9863d.  There's a bit more comment a few lines higher on the larger logic between the split of stages as well.